packages/esbuild/src/executors/esbuild/lib/normalize.ts

```diff
@@ -18,6 +18,9 @@ export function normalizeOptions(
       ...rest,
       external: options.external ?? [],
       singleEntry: false,
+      // Use the `main` file name as the output file name.
+      // This is needed for `@nrwl/js:node` to know the main file to execute.
+      outputFileName: `${parse(options.main).name}.js`,
```

Comment:
Do we need to think about the format here for the extension Ex. `.cjs` ?

Reply:
Yes we do, it is handled later on in `getOutfile`. I've added a comment here to explain. 

No matter what the user (or we) sets as the output file, it is set to the correct extension when calling esbuild and emitting `outfile` in `BuildEvent`.

A bit unintuitive but it works.